src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,36 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception ex) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Failed to save the movie. " + ex.getMessage());
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        Session session;
+        try {
+            session = factory.openSession();
+            Optional<Movie> result = Optional.ofNullable(session.get(Movie.class, id));
+            session.close();
+            return result;
+        } catch (Exception ex) {
+            throw new DataProcessingException("Failed to create a session" + ex.getMessage());
```

Comment:
Consider including the original exception as a cause in the `DataProcessingException` to preserve the stack trace. This can be done by using `throw new DataProcessingException("Failed to create a session", ex);`.